.github/workflows/compile-examples.yml

```diff
@@ -0,0 +1,265 @@
+              - libraries/I2S/examples
+              - libraries/SDU/examples
+          # Vidor board
+          - board:
+              type: "vidor"
+            additional-sketch-paths: |
+              - libraries/SAMD_BootloaderUpdater/examples
+              - libraries/I2S/examples
+          # mkrwifi1010
+          - board:
+              type: "mkrWiFi1010"
+            additional-sketch-paths: |
+              - libraries/I2S/examples
+              - libraries/SDU/examples
+              - ~/Arduino/libraries/WiFiNINA/examples
+          # mkrnb1500 boards
+          - board:
+              type: "mkrNB1500"
+            additional-sketch-paths: |
+              - libraries/I2S/examples
+              - libraries/SDU/examples
+          # nano_33_iot
+          - board:
+              type: "nano33IoT"
+            additional-sketch-paths: |
+              - libraries/I2S/examples
+              - ~/Arduino/libraries/WiFiNINA/examples
+          # mkrwan1310
+          - board:
+              type: "mkrwan1310"
+            additional-sketch-paths: |
+              - libraries/I2S/examples
+          # adafruit_cicrcuitplayground board
+          - board:
+              type: "adafruit_playg"
+            additional-sketch-paths: |
+              - libraries/SDU/examples
+          # mzero boards
+          - board:
+              type: "mzero"
+            additional-sketch-paths: |
+              - ~/Arduino/libraries/Firmata/examples/StandardFirmataPlus
+              - ~/Arduino/libraries/Firmata/examples/StandardFirmataEthernet
+              - ~/Arduino/libraries/Firmata/examples/StandardFirmata
+              - ~/Arduino/libraries/Firmata/examples/SimpleDigitalFirmata
+              - ~/Arduino/libraries/Firmata/examples/SimpleAnalogFirmata
+              - ~/Arduino/libraries/Firmata/examples/ServoFirmata
+              - ~/Arduino/libraries/Firmata/examples/EchoString
+              - ~/Arduino/libraries/Firmata/examples/AnalogFirmata
+              - ~/Arduino/libraries/Firmata/examples/AllInputsFirmata
+              - libraries/SDU/examples
+          # tian board
+          - board:
+              type: "tian"
+            additional-sketch-paths: |
+              - ~/Arduino/libraries/Firmata/examples/StandardFirmataPlus
+              - ~/Arduino/libraries/Firmata/examples/StandardFirmataEthernet
+              - ~/Arduino/libraries/Firmata/examples/StandardFirmata
+              - ~/Arduino/libraries/Firmata/examples/SimpleDigitalFirmata
+              - ~/Arduino/libraries/Firmata/examples/SimpleAnalogFirmata
+              - ~/Arduino/libraries/Firmata/examples/ServoFirmata
+              - ~/Arduino/libraries/Firmata/examples/EchoString
+              - ~/Arduino/libraries/Firmata/examples/AnalogFirmata
+              - ~/Arduino/libraries/Firmata/examples/AllInputsFirmata
+          # MKRGSM1400 board
+          - board:
+              fqbn: "arduino:samd:mkrgsm1400"
+            additional-sketch-paths: |
```

Comment:
Since you set `matrix.board.type` to `usb` for this board, I suspect you intended for the paths under `matrix.include[].board.type.usb" to be compiled for it, but what is happening is the value of the `matrix.board.additional-sketch-paths` key is being overwritten. So it only ends up containing the single path `~/Arduino/libraries/MKRGSM/examples`, as you can see from this workflow log;
https://github.com/giulcioffi/ArduinoCore-samd/runs/1263161053?#step:7:16602

You can fix this by using a dedicated key for each, like `matrix.board.additional-usb-sketch-paths`  and `matrix.board.additional-sketch-paths` . You can see an example of that here:
https://github.com/arduino/arduino-examples/blob/main/.github/workflows/compile-examples.yml